### ISSUE TITLE
Why Val_loss not equal val_regresssion + val_classification

### ISSUE THREAD

**Issue by Mixelon-tera** [fizyr/keras-retinanet#1320]
<img width="737" alt="Screen Shot 2563-04-02 at 17 53 41" src="https://user-images.githubusercontent.com/47152607/78242327-04def780-750c-11ea-85d5-76c6182e9188.png">

I found that train_loss = train_regression_loss + train_classification_loss.
But for validation, the val_loss is larger than val_regression_loss + val_classification_loss.
Why is it like this?

**Comment by hgaiser** [comment]
Hah good question, I never noticed that before. Though we outsource computing of the val_loss to keras, so the issue is really coming from there.

**Comment by ishouldbedany** [comment]
This only seems to happen in the validation set, however. As seen above, for the current training batch, `loss` = `regression_loss` + `classification_loss`. 

My experience has been similar.

**Comment by rubeenmp** [comment]
Hi, how can I obtain the evaluation loss? Does it also appear on tensorboard?

**Comment by chrisparker126** [comment]
did a bit of research and as @hgaiser said its looks to be coming from keras:
When running evaluate, keras computes a **running** mean and from what I can see it accumulates quite a bit of error. If you compute the per batch loss manually (using test_on_batch) and average over reg and class loss add up meaninfully 
Not sure how to solve this. perhaps force keras to use float64?? Tried this in keras/metrics.py but seems the error builds up to quickly.
